program/src/register_mint.rs

```diff
@@ -0,0 +1,89 @@
+use jito_bytemuck::AccountDeserialize;
+use jito_jsm_core::loader::load_system_account;
+use jito_restaking_core::{config::Config, ncn::Ncn, ncn_vault_ticket::NcnVaultTicket};
+use jito_tip_router_core::{
+    error::TipRouterError, tracked_mints::TrackedMints, weight_table::WeightTable,
+};
+use jito_vault_core::{vault::Vault, vault_ncn_ticket::VaultNcnTicket};
+use solana_program::{
+    account_info::AccountInfo,
+    entrypoint::ProgramResult,
+    msg,
+    program_error::ProgramError,
+    pubkey::Pubkey,
+    system_program,
+    sysvar::{clock::Clock, Sysvar},
+};
+
+pub fn process_register_mint(program_id: &Pubkey, accounts: &[AccountInfo]) -> ProgramResult {
+    let [restaking_config, tracked_mints, ncn, weight_table, vault, vault_ncn_ticket, ncn_vault_ticket, restaking_program_id, vault_program_id] =
+        accounts
+    else {
+        return Err(ProgramError::NotEnoughAccountKeys);
+    };
+
+    TrackedMints::load(program_id, ncn.key, tracked_mints, true)?;
+    Ncn::load(restaking_program_id.key, ncn, false)?;
+    Vault::load(vault_program_id.key, vault, false)?;
+    VaultNcnTicket::load(vault_program_id.key, vault_ncn_ticket, vault, ncn, false)?;
+    NcnVaultTicket::load(
+        restaking_program_id.key,
+        ncn_vault_ticket,
+        ncn,
+        vault,
+        false,
+    )?;
+
+    let epoch_length = {
+        let restaking_config_data = restaking_config.data.borrow();
+        Config::try_from_slice_unchecked(&restaking_config_data)?.epoch_length()
+    };
+
+    let slot = Clock::get()?.slot;
```

Comment:
We use the restaking_config's `epoch_length` to calculate the ncn_epoch. Take a look at how the weight table gets it.

Reply:
good catch

Reply:
wait that is what this is doing right